spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -65,6 +65,13 @@ object DynamicInterpreter {
     conformedDf
   }
 
+  def findRulesBreakingImmutability(steps: List[ConformanceRule], schema: StructType): Seq[DropConformanceRule] = {
+    steps.flatMap {
+      case rule: DropConformanceRule => if (SchemaUtils.fieldExists(rule.outputColumn, schema)) Some(rule) else None
```

Comment:
```suggestion
      case rule: ConformanceRule => if (SchemaUtils.fieldExists(rule.outputColumn, schema)) Some(rule) else None
```
My thinking here is that any (existing or future) conformance rule may be in theory breaking the immutability of the original columns, not specifically the `DropConformanceRule` in general (even though maybe it is true now). Thus, the problematic rules could be defined more broadly and in a future-proof way, so this is my attempt to do so.

I am basically hinting [Uncle Bob's Avoid Switch Statement clause](https://medium.com/@myoungsokang_36227/uncle-bob-says-avoid-switch-statements-63d94e3b59e9).

Reply:
transformed into anonymous function